community/kernel/src/main/java/org/neo4j/kernel/impl/store/record/RelationshipRecord.java

```diff
@@ -238,13 +238,15 @@ public boolean equals( Object o )
                secondPrevRel == that.secondPrevRel &&
                secondNextRel == that.secondNextRel &&
                firstInFirstChain == that.firstInFirstChain &&
-               firstInSecondChain == that.firstInSecondChain;
+               firstInSecondChain == that.firstInSecondChain &&
+               this.requiresSecondaryUnit() == that.requiresSecondaryUnit() &&
```

Comment:
I don't think secondary unit ID should be included in comparisons, because that's not really part of the logical record data... same as with f.ex. `created`. Secondary unit ID has more to do with the format than with the data. The reason it's in here is that we need to have this information in the log when we write that and records are the main carriers of data in and around committing transactions.